sql/src/main/java/io/crate/expression/reference/sys/check/cluster/TablesNeedRecreationSysCheck.java

```diff
@@ -61,30 +63,38 @@ public BytesRef description() {
 
     @Override
     public boolean validate() {
-        tablesNeedRecreation = tablesNeedRecreation(clusterService.state().metaData());
-        return tablesNeedRecreation.isEmpty();
+        Collection<TableIdent> needsRecreation = tablesNeedRecreation(clusterService.state().metaData());
+        tablesNeedRecreation = needsRecreation
+            .stream()
+            .map(TableIdent::sqlFqn)
+            .collect(Collectors.toList());
+        return needsRecreation.isEmpty();
     }
 
     // We need to check for the version that the index was created since the lucene segments might
     // be automatically upgraded to the latest version (can happen when data was in the translog) so
     // minimumCompatVersion cannot be used to indicate that a table recreation is needed.
     @VisibleForTesting
-    static boolean isRecreationRequired(String indexName, IndexMetaData indexMetaData) {
-        return indexMetaData != null && !BlobIndex.isBlobIndex(indexName) && indexMetaData.getCreationVersion().before(org.elasticsearch.Version.V_5_0_0);
+    static boolean isRecreationRequired(IndexMetaData indexMetaData) {
+        return indexMetaData != null && indexMetaData.getCreationVersion().before(org.elasticsearch.Version.V_5_0_0);
     }
 
     /**
      * Retrieves an ordered collection of table FQNs that need
      * to be recreated to be compatible with future CrateDB versions.
      * @return the ordered collection of table FQNs that need to be recreated
      */
-    static Collection<String> tablesNeedRecreation(MetaData clusterIndexMetaData) {
-        Collection<String> tablesNeedRecreation = new TreeSet<>();
+    static Collection<TableIdent> tablesNeedRecreation(MetaData clusterIndexMetaData) {
+        Collection<TableIdent> tablesNeedRecreation = new TreeSet<>();
         for (ObjectObjectCursor<String, IndexMetaData> entry : clusterIndexMetaData.indices()) {
             String indexName = entry.key;
             IndexMetaData indexMetaData = entry.value;
-            if (isRecreationRequired(indexName, indexMetaData)) {
-                tablesNeedRecreation.add(TableIdent.fromIndexName(indexName).fqn());
+            if (isRecreationRequired(indexMetaData)) {
+                if (BlobIndex.isBlobIndex(indexName)) {
```

Comment:
seems like `TableIdent.fromIndexName` doesn't work correctly for blob tables - I'll look into that separately on master.